pkg/nitrite/nitrite_test.go

```diff
@@ -0,0 +1,128 @@
+package nitrite_test
+
+import (
+	_ "embed"
+	"encoding/base64"
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/blocky/nitrite/pkg/attestation"
+	"github.com/blocky/nitrite/pkg/nitrite"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+//go:embed test/nitro-stag-attestation.txt
+var nitroStagAttestBase64 string
+
+//go:embed test/nitro-stag-attestation-doc.json
+var nitroStagAttestDocJSON string
+
+func makeAttestationDocFromJSON(
+	attestationDocJSON []byte,
+) (*attestation.Document, error) {
+	doc := attestation.Document{}
+	err := json.Unmarshal(attestationDocJSON, &doc)
+	if err != nil {
+		return nil, err
+	}
+	if doc.UserData == nil {
+		doc.UserData = make([]byte, 0)
+	}
+	if doc.Nonce == nil {
+		doc.Nonce = make([]byte, 0)
+	}
+	return &doc, nil
+}
+
+func makeNitroStagVerifyOptions(
+	currentTime time.Time,
+) nitrite.VerifyOptions {
+	return nitrite.VerifyOptions{
+		Roots:               nil,
+		CurrentTime:         currentTime,
+		AllowSelfSignedCert: false,
+	}
+}
+
+func TestNitrite_Verify(t *testing.T) {
```

Comment:
Reimplement `Verify` as `VerifyFromRaw` (called by `Verify`) so that you can pass in mocks to `VerifyFromRaw`. As it is, you're not actually testing all the failure cases in `Verify`.